File: NativeBinderDemo/ClientDemo.cpp
#include "IMyService.h"

int main() {
	sp < IServiceManager > sm = defaultServiceManager(); //获取service manager引用
	sp < IBinder > binder = sm->getService(String16("service.myservice"));//获取名为"service.myservice"的binder接口
	sp<IMyService> cs = interface_cast < IMyService > (binder);//将biner对象转换为强引用类型的IMyService
	cs->sayHello();//利用binder引用调用远程sayHello()方法
	return 0;
}

File: FrameworkBinderDemo/ClientDemo/src/com/yuanhh/frameworkBinder/ClientDemo.java
package com.yuanhh.frameworkBinder;

import android.os.IBinder;
import android.os.RemoteException;
import android.os.ServiceManager;

public class ClientDemo {
 
	public static void main(String[] args) throws RemoteException {
		System.out.println("Client start");
		IBinder binder = ServiceManager.getService("MyService"); //获取名为"MyService"的服务
		IMyService myService = new MyServiceProxy(binder); //创建MyServiceProxy对象
		myService.sayHello("binder"); //通过MyServiceProxy对象调用接口的方法
		System.out.println("Client end");
	}
}
